src/render.rs

```diff
@@ -2697,3 +2798,793 @@ unsafe fn copy_buffer<T: std::marker::Copy>(
+        .build();
+
+    let descriptor_writes = [
+        vd::WriteDescriptorSet::builder()
+            .dst_set(descriptor_set)
+            .dst_binding(0)
+            .dst_array_element(0)
+            .descriptor_count(1)
+            .descriptor_type(vd::DescriptorType::UniformBuffer)
+            .buffer_info(&shared_info)
+            .build(),
+        vd::WriteDescriptorSet::builder()
+            .dst_set(descriptor_set)
+            .dst_binding(1)
+            .dst_array_element(0)
+            .descriptor_count(1)
+            .descriptor_type(vd::DescriptorType::CombinedImageSampler)
+            .image_info(&tex_descriptor)
+            .build()
+    ];
+
+    _descriptor_pool.update_descriptor_sets(&descriptor_writes, &[]);
+
+    let resources = init_text_pipeline_resources(extent2d);
+
+    Ok(
+        TextMeta{
+            buffer_3d,
+            memory_3d,
+            _image,
+            image_memory,
+            _view,
+            _descriptor_pool,
+            _descriptor_set_layout,
+            descriptor_set,
+            _sampler,
+            pipeline_layout,
+            resources,
+        }
+    )
+}
+
+#[derive(Clone, PartialEq)]
+pub enum TextScale {
+    ScreenSpace,
+    ScaleFactor,
+}
+
+#[derive(Clone)]
+pub struct Text {
+    pub text: String,
+    pub position: alg::Vec3,
+    pub align: TextAlign,
+    pub scale: TextScale,
+    pub scale_factor: f32,
+    pub is_2d: bool,
+}
+
+impl Text {
+	pub fn empty_2d_instance() -> Text {
+		Text {
+			text: "".to_string(),
+			position: alg::Vec3::zero(),
+            align: TextAlign::AlignCenter,
+            scale: TextScale::ScreenSpace,
+            scale_factor: 1f32,
+            is_2d: true,
+		}
+	}
+
+    pub fn empty_3d_instance() -> Text {
+        Text {
+            text: "".to_string(),
+            position: alg::Vec3::zero(),
+            align: TextAlign::AlignRight,
+            scale: TextScale::ScaleFactor,
+            scale_factor: 1f32,
+            is_2d: false,
+        }
+    }
+}
+
+struct TextDisplay {
+    _framebuffer_width: u32,
+    _framebuffer_height: u32,
+    device: vd::Device,
+    pipeline: vd::GraphicsPipeline,
+	num_letters: u64,
+    is_3d: bool,
+}
+
+const TEXTOVERLAY_MAX_CHAR_COUNT:u32 =  2048;
+
+fn create_text(
+    vulkan_device: vd::Device,
+    assembly: vd::PipelineInputAssemblyStateCreateInfo,
+    multisampling: vd::PipelineMultisampleStateCreateInfo,
+    render_pass: &vd::RenderPass,
+    text_meta: &TextMeta,
+    is_3d: bool,
+) -> vd::Result<
+        TextDisplay,
+    > {
+    let resources = &text_meta.resources;
+    
+    // Loading shaders
+    let path = {
+        let mut path = &config::load_section_setting::<String>(
+            &config::ENGINE_CONFIG,
+            "settings",
+            "shader_path"
+        );
+
+        [path, "/"].concat()
+    };
+
+    let type_shader = if is_3d {"3"} else {"2"};
+    
+    println!("Loading {}d text shaders from \"{}\"", type_shader, path);
```

Comment:
If you're going to add this message, we should probably add one for the debug shaders as well